### ISSUE TITLE
[JENKINS-53501] Render sequential stages that are not in parallel

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4852]
<p>The new sequential stages improvement is awesome and I understand it's meant for use in parallel, but I do have a use case where I am not using it in parallel. </p>

<p>Take this pipeline example for integration tests:</p>

<pre>
stage(<span class="code-quote">'Integration Tests'</span>) {
    options {
lock(<span class="code-quote">'ci-instrument'</span>)
    }
    stages {
stage(<span class="code-quote">'Setup'</span>) {
    agent {
label <span class="code-quote">'ci-instrument'</span>
    }
    steps {
sh <span class="code-quote">'<setup steps>'</span>
    }
stage(<span class="code-quote">'Run Tests'</span>) {
    agent {
dockerfile {

}
    }
    steps {
       sh 'NOT running on <span class="code-quote">"ci-instrument"</span>, but it still needs to be locked"
    }
}
stage(<span class="code-quote">'Copy results'</span>) {
 ....
}
    }
}</pre>
<p> </p>

<p>This is visualized as <br/>
<span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/44434/44434_image-2018-09-10-14-04-00-895.png" style="border: 0px solid black" /></span></p>

<p>It would be nice to have the Setup, Run Tests, and Copy Test Results stages to be under the Integration Tests stage, either always visible or maybe clicking Integration Tests will pop up the stages under it. </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilederman.jpg" title="ilederman's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ilederman">ilederman</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53501" target="_blank">Render sequential stages that are not in parallel</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: blueocean-backend, blueocean-ui
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 5
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The new sequential stages improvement is awesome and I understand it's meant for use in parallel, but I do have a use case where I am not using it in parallel. </p>

<p>Take this pipeline example for integration tests:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">stage(<span class="code-quote">'Integration Tests'</span>) {
    options {
        lock(<span class="code-quote">'ci-instrument'</span>)
    }
    stages {
        stage(<span class="code-quote">'Setup'</span>) {
            agent {
                label <span class="code-quote">'ci-instrument'</span>
            }
            steps {
                sh <span class="code-quote">'&lt;setup steps&gt;'</span>
            }
        stage(<span class="code-quote">'Run Tests'</span>) {
            agent {
                dockerfile {

                }
            }
            steps {
               sh 'NOT running on <span class="code-quote">"ci-instrument"</span>, but it still needs to be locked"
            }
        }
        stage(<span class="code-quote">'Copy results'</span>) {
         ....
        }
    }
}</pre>
</div></div>
<p> </p>

<p>This is visualized as 
<span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/44434/44434_image-2018-09-10-14-04-00-895.png" style="border: 0px solid black" /></span></p>

<p>It would be nice to have the Setup, Run Tests, and Copy Test Results stages to be under the Integration Tests stage, either always visible or maybe clicking Integration Tests will pop up the stages under it. </p></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [image-2018-09-10-14-04-00-895.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/44434/image-2018-09-10-14-04-00-895.png)
  > ![image-2018-09-10-14-04-00-895.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/44434/image-2018-09-10-14-04-00-895.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ragnellr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53501?focusedId=360341&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-360341">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The "Integration Tests" stage in Ilya's example is really a "wrapper" stage that should be displayed as such. It'd be nice if its progress could be shown as the cumulative progress of all of its "child" stages, displayed somewhere above all of the individual stages that fall under it.</p>
</pre>
</details>

</ul>
</details>

<p>The "Integration Tests" stage in Ilya's example is really a "wrapper" stage that should be displayed as such. It'd be nice if its progress could be shown as the cumulative progress of all of its "child" stages, displayed somewhere above all of the individual stages that fall under it.</p>
